grafana/iot-sitewise-datasource
# Add support for unassociated streams

**What this PR does / why we need it**:
When querying by propertyAlias was first implemented, it had support for unassociated streams ("disassociated" in AWS). 
Unassociated streams are data streams that are not associated with a propertyId on an asset, but contain historical data about a property all the same. They can be queried by a propertyAlias. 
You can see them here in the console if you apply the Disassociated Streams filter: 

<img width="800" alt="Screenshot 2023-10-13 at 7 13 43 PM" src="https://github.com/grafana/iot-sitewise-datasource/assets/16140639/bcfb6115-7b8a-4552-9e71-ffeb965d46f4">

[The 500 error from the bug report](https://github.com/grafana/iot-sitewise-datasource/issues/228) happened because of a null pointer reference since, when we query DescribeAssetPropertyWithContext for unassociated propertyAliases, it doesn't return an assetId or propertyId. The bug was introduced when we implemented fetching assets and property values only by propertyAlias. However, this only worked for associated streams, where DescribeAssetPropertyWithContext will actually return propertyId.

```
func getAssetIdAndPropertyId(query models.AssetPropertyValueQuery, client client.SitewiseClient, ctx context.Context) (models.AssetPropertyValueQuery, error) {
	result := query
	if query.PropertyAlias != "" {
		resp, err := client.DescribeTimeSeriesWithContext(ctx, &iotsitewise.DescribeTimeSeriesInput{
			Alias: getPropertyAlias(query.BaseQuery),
		})
		if err != nil {
			return models.AssetPropertyValueQuery{}, err
		}
		---> result.AssetIds = []string{*resp.AssetId}
		result.PropertyId = *resp.PropertyId
		
```
<img width="1422" alt="Screenshot 2023-10-16 at 12 13 14 PM" src="https://github.com/grafana/iot-sitewise-datasource/assets/16140639/c9f6fbff-0f9a-442a-af38-c03d8502267e">


On the frontend, I fixed the panel header to show propertyAlias in the panel header when collapsed
<img width="948" alt="Screenshot 2023-10-16 at 12 11 40 PM" src="https://github.com/grafana/iot-sitewise-datasource/assets/16140639/d7996deb-95ba-48e1-be12-d0054650697c">

I also moved some functions in the util package from /api, since I needed them in other packages

**Which issue(s) this PR fixes**:


### How to test: 
Add the following propertyAliased into the property alias field and test different "Get value ..." query types. For data, select "last year" in order to be able to see data, then you can zoom in. 

We have a few aliased streams in our dev account: 
turbine/1/torque -> disassociated - doesn't have an assetId/propertyId tied to it, so this can be tested for a fix
/turbine/1/wind_direction -> associated - for comparison


Fixes #

**Special notes for your reviewer**:
